types/got/index.d.ts

```diff
@@ -131,13 +137,18 @@ declare namespace got {
         request?: number;
     }
 
+    interface RetryOptions {
+        retries?: number;
```

Comment:
Property `retries` can be function like `retry` in `got v8`. See the docs https://github.com/sindresorhus/got#retry

> The retries property can be a number or a function with retry and error arguments. The function must return a delay in milliseconds (0 return value cancels retry).